net2cog/netcdf_convert.py

```diff
@@ -166,36 +234,28 @@ def netcdf_converter(
     netcdf_file = os.path.abspath(input_nc_file)
     logger.debug('NetCDF Path: %s', netcdf_file)
 
-    if netcdf_file.endswith('.nc'):
+    if netcdf_file.endswith(('.nc', '.nc4', 'h5')):
```

Comment:
Let me know if `.h5` is too permissive here. (It allowed me to test SPL4CMDL, but in reality they might be coming through HOSS first, and so be `.nc4`)

Reply:
Technically speaking it _probably_ is too permissive.

We have a dependency on `netcdf4` python library so I believe xarray will use that as the backend. Since not all H5 files are valid NetCDF4, there is probably a small subset of files that wouldn't work.

Theoretically we could try using a different backend for xarray that could handle both; is xarray smart enough to pick the right one if we had multiple backends installed?

As far as this PR is concerned I think it is worth accepting this risk in order to provide better functionality at this time.

Reply:
In other words, if we run into a situation where this is a problem we can handle it as a bug and make a decision at that time.